smartsim/entity/files.py

```diff
@@ -78,12 +80,9 @@ def _check_files(self):
         self.copy = self._type_check_files(self.copy, "Copyable")
         self.link = self._type_check_files(self.link, "Symlink")
 
-        # check the paths provided by the user and ensure
-        # that no directories were provided as tagged files
-        for i in range(len(self.tagged)):
-            self.tagged[i] = self._check_path(self.tagged[i])
-            if path.isdir(self.tagged[i]):
-                raise SSConfigError("Directories cannot be listed in tagged files")
+        self.tagged_hierarchy = TaggedFilesHierarchy.from_list_paths(
```

Comment:
Does this preserve the directory structure?

Reply:
It does! `TaggedFilesHierarchy.from_list_paths` will create and return a tree of `TaggedFilesHierarchy`s with a list of path to files at each node. By DFSing over this structure in the `Generator` and creating a new directory for each TaggedFilesHierarchy and copying in each file in its list of files, we are able to preserve the structure of any dirs the `self.tagged` list